macros/src/main/scala/me/rexim/morganey/meta/QuotationMacro.scala

```diff
@@ -134,23 +124,16 @@ private[meta] class QuotationMacro(val c: Context) {
         (q"_root_.scala.Some((..$ys))", q"_root_.scala.None")
     }
 
-    if (!hasSubpatterns() && ps.size == 1) {
-      q"""
-        new {
-          def unapply(input: $LambdaTermTpe) = Option(input)
-        }.unapply(..$args)
-      """
-    } else {
-      q"""
-        new {
-          def unapply(input: $LambdaTermTpe) = input match {
-            case $term => $ifp
-            case _     => $elp
-          }
-        }.unapply(..$args)
-      """
-    }
-
+    val identName = TermName(c.freshName())
+    q"""
+      new {
+        val $identName = true
```

Comment:
> Could you please explain the purpose of `$identName` here?

Of course! 
It's a workaround for the warning noticed by @rexim.

> I remove that and the tests show the warning:

As you might notice, the error only appears on patterns, which match the whole ast.
In this case the macro expands to something like:
```scala
({
final class $anon {
  def unapply(input: me.rexim.morganey.ast.LambdaTerm) = input match {
    case (x0 @ _) => _root_.scala.Some(x0)
    case _ => _root_.scala.None
  }
};
new $anon()
}).unapply(<unapply-selector>)
```

This causes the warning, because the first pattern matches every expression.

> Unfortunately, I have no idea how to apply backticks to macro code. That's probably the reason for this non-obvious clause.

The backticks won't fix the problem, because a variable has to be use to match the whole tree.